Source/JavaScriptCore/runtime/IntlDurationFormat.cpp

```diff
@@ -575,7 +575,7 @@ static Vector<Element> collectElements(JSGlobalObject* globalObject, const IntlD
                 bool needsFormatMinutes = (needsFormatHours && needsFormatSeconds) || duration[TemporalUnit::Minute] || durationFormat->units()[static_cast<unsigned>(TemporalUnit::Minute)].display() != IntlDurationFormat::Display::Auto;
 
                 bool needsFormat = (unit == TemporalUnit::Hour && needsFormatHours) || (unit == TemporalUnit::Minute && needsFormatMinutes) || (unit == TemporalUnit::Second && needsFormatSeconds);
-                bool needsSeparator = (unit == TemporalUnit::Hour && needsFormatMinutes) || (unit == TemporalUnit::Minute && needsFormatSeconds);
+                bool needsSeparator = (unit == TemporalUnit::Hour && needsFormatHours && needsFormatMinutes) || (unit == TemporalUnit::Minute && needsFormatSeconds);
```

Comment:
just to clarify the flow, will it fix this such a case?
```
const df = new Intl.DurationFormat('en-GB', {
  style: "digital",
  hoursDisplay: "auto"
});

df.format({ days: 1, hours: 0, minutes: 1, seconds: 2 }); // "1 day:01:02"
```

Currently there is still a separator:

Reply:
After this patch, `df.format({ days: 1, hours: 0, minutes: 1, seconds: 2 });` is `"1 day, 01:02"`. This behavior matches to the spec?

Reply:
I updated the test, so please see that.

Reply:
> After this patch, `df.format({ days: 1, hours: 0, minutes: 1, seconds: 2 });` is `"1 day, 01:02"`. This behavior matches to the spec?

yes, thanks